Update cache mediator docs (#1000)
wso2/docs-mi: en/docs/reference/mediators/cache-mediator.md

# Cache Mediator

When a message enters a message flow, the Cache mediator checks whether the incoming message is similar to a previous message that was received

> within a specified period of time. This is done by evaluating the hash value of the incoming message. 

Reviewer: ```suggestion
within a specified duration. This is done by evaluating the hash value of the incoming message. 
```

Let's use duration or period.

<table>
<thead>
<tr class="header">
<th></th>
<th>If a similar message was received before</th>
<th>If a similar message was not received before</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>If the <code>onCacheHit</code> sequence is configured</strong></td>

> <td>Cache mediator executes the onCacheHit sequence, fetches the cached response. The response will be sent if the <a href="{{base_path}}/reference/mediators/respond-mediator">respond mediator</a> is defined.</td>

Reviewer: ```suggestion
<td>The Cache mediator executes the <code>onCacheHit</code> sequence and fetches the cached response. The response will be sent if the <a href="{{base_path}}/reference/mediators/respond-mediator">respond mediator</a> is defined.</td>
```

<td>The message is passed on to the subsequent mediators.</td>
</tr>
<tr class="odd">
<td><strong>If the <code>onCacheHit</code> sequence is not configured</strong></td>

> <td>The cached response is sent back to the client and the message is not passed on.</td>

Reviewer: ```suggestion
<td>The cached response is returned to the client and the message is not passed on.</td>
```

    
## Configuration

> **Cache type** specifies whether the Cache mediator should be in the incoming path (to check the request) or in the outgoing path (to cache the response). Possible values are as follows.

Reviewer: ```suggestion
The **Cache type** specifies whether the Cache mediator should be in the incoming path (to check the request) or the outgoing path (to cache the response). Possible values are as follows:
```

</tr>
<tr class="even">
<td><strong>Protocol Type</strong></td>

> <td>The protocol type to be cached in the message flow. In the current implementation, HTTP is the only value that you can select. Although the only configuration supported for other protocols is the <code>HashGenerator</code>, you can specify the protocol type to be anything and specify a <code>                HashGenerator               </code> that you prefer.</td>

Reviewer: ```suggestion
<td>The protocol type to be cached in the message flow. In the current implementation, HTTP is the only value you can select. Although the only configuration supported for other protocols is the <code>HashGenerator</code>, you can specify the protocol type to be anything and specify a <code>HashGenerator</code> you prefer.</td>
```

</tr>
<tr class="odd">
<td><strong>HTTP Methods</strong></td>

> <td>A comma separated list of HTTP methods that should be cached for the HTTP protocol. The default value is <code>*</code>, where it caches all HTTP methods.</td>

Reviewer: ```suggestion
<td>A comma-separated list of HTTP methods that should be cached for the HTTP protocol. The default value is <code>*</code>, where it caches all HTTP methods.</td>
```

</tr>
<tr class="even">
<td><strong>Headers to Exclude in Hash</strong></td>

> <td>A comma separated list of headers to ignore when hashing an incoming messages. If you want to exclude all headers when hashing an incoming message, specify <code>*</code>.</td>

Reviewer: ```suggestion
<td>A comma-separated list of headers to ignore when hashing an incoming messages. If you want to exclude all headers when hashing an incoming message, specify <code>*</code>.</td>
```

<td><div class="content-wrapper">
<p>This parameter is used to define the logic used by the Cache mediator to evaluate the hash values of incoming messages. The value specified here should be a class that implements the <code>org.separated.carbon.mediator.cache.digest.DigestGenerator</code> class interface. The default hash generator is <code>org.wso2.carbon.mediator.cache.digest.HttpRequestHashGenerator</code>. If the generated hash value is found in the cache, then the Cache mediator executes the <code>onCacheHit</code> sequence, which can be specified inline or referenced.</p>
<div class="admonition note">
    <p class="admonition-title">Note</p>
    <p>The default hash generator is specific to the HTTP protocol.</p>
    <p>If you are using any other protocol, you need to write a custom hash generator or use one of the following deprecated hash generator classes:</p>
        <ul>
        <li><code>org.wso2.carbon.mediator.cache.digest.DOMHASHGenerator</code></li>
        <li><code>org.wso2.carbon.mediator.cache.digest.REQUESTHASHGenerator</code></li>
        </ul>
</div>

</div></td>
</tr>
<tr class="odd">
<td><strong>Enable Cache Control Headers</strong></td>

> <td><p>Whether the Cache mediator should honor the Cache-Control header(no-cache, no-store, max-age headers). If you set this to <code>false</code> which is the default value, the Cache mediator will not consider the Cache-Control headers when caching the response or when returning the cached response.</p>

Reviewer: ```suggestion
<td><p>Whether the Cache mediator should honor the Cache-Control header (no-cache, no-store, max-age headers). If you set this to <code>false</code> which is the default value, the Cache mediator will not consider the Cache-Control headers when caching the response or returning the cached response.</p>
```

<td>The maximum number of elements to be cached. The default size is 1000.</td>
</tr>
<tr class="even">
<td><strong>Sequence Type</strong></td>
<td>
<ul>

> <li><strong>Anonymous</strong> : If this option is selected, an anonymous sequence is executed in cache hit.</li>

Reviewer: ```suggestion
<li><strong>Anonymous</strong> : If this option is selected, an anonymous sequence is executed in a cache hit.</li>
```

> <li><strong>Registry reference</strong> : If this option is selected, the reference to the <code>onCacheHit</code> sequence has to be configured in the <strong>Sequence Key</strong> field. The sequence should be created in the registry in order to be specified in this field.</li>

Reviewer: ```suggestion
<li><strong>Registry reference</strong> : If this option is selected, the reference to the <code>onCacheHit</code> sequence has to be configured in the <strong>Sequence Key</strong> field. The sequence should be created in the registry to be specified in this field.</li>
```
